ui/apps/dashboard/src/app/(organization-active)/(dashboard)/env/[environmentSlug]/functions/[slug]/InvokeButton/InvokeModal.tsx

```diff
@@ -0,0 +1,92 @@
+import { useState } from 'react';
+import { Button } from '@inngest/components/Button';
+import { Modal } from '@inngest/components/Modal';
+
+import CodeEditor from '@/components/Textarea/CodeEditor';
+
+const initialCode = { data: {} };
+
+type Props = {
+  isOpen: boolean;
+  onCancel: () => void;
+  onConfirm: (payload: { data: Record<string, unknown> }) => void;
+};
+
+export function InvokeModal({ isOpen, onCancel, onConfirm }: Props) {
+  const [error, setError] = useState<string>();
+
+  function onSubmit(event: React.FormEvent<HTMLFormElement>) {
+    event.preventDefault();
+    const formData = new FormData(event.currentTarget);
+
+    try {
+      const payload = parseCode(formData.get('code'));
+      onConfirm(payload);
+      setError(undefined);
+    } catch (error) {
+      if (!(error instanceof Error)) {
+        setError('Unknown error');
+        return;
+      }
+
+      setError(error.message);
+    }
+  }
+
+  return (
+    <Modal
+      className="w-[800px]"
+      description="Invoke this function, triggering a function run"
+      isOpen={isOpen}
+      onClose={onCancel}
+      title={<h2 className="mb-4 text-lg font-medium">Invoke Function</h2>}
+    >
+      <form onSubmit={onSubmit}>
+        <div className="border-b border-slate-200 p-6">
+          <CodeEditor
+            className="rounded-lg bg-slate-900 px-4"
+            initialCode={JSON.stringify(initialCode, null, 2)}
+            language="json"
+            name="code"
+          />
+
+          {error && <div className="pt-4 text-red-500">{error}</div>}
```

Comment:
```suggestion
          {error && <div className="pt-4 text-rose-700">{error}</div>}
```

Assuming this is in a white background, we usually go for rose-700. On dark backgrounds I think it will be rose-500